packages/ui/src/components/Lessons/LessonCard.tsx

```diff
@@ -225,12 +229,30 @@ export const LessonCard: React.FC<Props> = ({
     else return intl("lessons.button.send-message");
   }, [canceled, currentUserRole, intl]);
 
+  const disabledBtn = useMemo(() => {
+    if (mainBtnHandler === onJoin && !canJoin) return true;
+    if (mainBtnHandler === onRebook && !canRebook) return true;
+    if (mainBtnHandler === onSendMsg && sendingMessage) return true;
+    if (sendMsgBtnDisabled) return true;
+    return false;
+  }, [
+    mainBtnHandler,
+    canJoin,
+    canRebook,
+    sendMsgBtnDisabled,
+    onJoin,
+    onRebook,
+    onSendMsg,
+    sendingMessage,
+  ]);
```

Comment:
It is hard on me to believe that this is a valid code. I believe you cannot check for equality on functions.